src/com/google/enterprise/adaptor/CommandStreamParser.java

```diff
@@ -467,6 +467,19 @@ public void readFromRetriever(DocId docId, Response response) throws IOException
     command = readCommand();
     while (command != null) {
       switch (command.getOperation()) {
+        case CONTENT:
+        case UP_TO_DATE:
+        case NOT_FOUND:
+          // Finish by putting accumulated ACL into response.
+          if (sendAclWithDocument) {
+            aclBuilder.setPermits(permits);
+            aclBuilder.setDenies(denies);
+            response.setAcl(aclBuilder.build());
+          }
+          break;
+        default:
+      }
+      switch (command.getOperation()) {
```

Comment:
Ah, I was reading this code wrong.  In the CL review, I was thinking somehow that getOperation() also does a read.  Which is completely the wrong reading.  

Can we do command.getOperation() once and use that gotten value in the two switches?

Something about the switch made me think it was always processing new token and that somehow we were processing two token at a time.  Which is completely the wrong reading, admitadly.